packages/pds/src/api/app/bsky/actor/putPreferences.ts

```diff
@@ -18,13 +18,19 @@ export default function (server: Server, ctx: AppContext) {
           throw new InvalidRequestError('Preference is missing a $type')
         }
       }
+
       await ctx.actorStore.transact(requester, async (actorTxn) => {
         await actorTxn.pref.putPreferences(
           checkedPreferences,
           'app.bsky',
           auth.credentials.scope,
         )
       })
+
+      await ctx.preferencesCache?.set(
+        requester,
+        JSON.stringify(checkedPreferences),
+      )
```

Comment:
I believe this could cache different contents depending on the user's scope.  Maybe it would be more straightforward to control the cache contents entirely on the read path as a read-through cache, and turn this into an invalidation.